### PR TITLE
Removing query-string parameters in `_analyze` API

### LINKED ISSUE #20246: standard analyzer bug when processing "number,number" ? [2.3.2]
**Elasticsearch version**: 2.3.2

**Plugins installed**: []

**JVM version**: sun 1.8.0_77

**OS version**: CentOS release 6.4 (Final)

**Description of the problem including expected versus actual behavior**:

**Steps to reproduce**:

1. create index

    ```
    POST testindex
    {
       "settings": {
          "index": {
             "number_of_shards": "1",
             "number_of_replicas": "0"
          }
       },
       "mappings": {
          "logs": {
             "dynamic": "false",
             "_all": {
                "enabled": false
             },
             "properties": {
                "snlist": {
                   "type": "string",
                   "analyzer": "standard"
                }
             }
          }
       }
    }
    ```

2. test analyzer with _analyze api

    ```
    GET_analyze?analyzer=standard&text=106614,106607'

    {
       "tokens": [
          {
             "token": "106614",
             "start_offset": 0,
             "end_offset": 6,
             "type": "<NUM>",
             "position": 0
          },
          {
             "token": "106607",
             "start_offset": 7,
             "end_offset": 13,
             "type": "<NUM>",
             "position": 1
          }
       ]
    }
    ```

3. post one document to testindex
    ```
    POST testindex/logs
    {
       "snlist": "106614,106607"
    }
    ```

4. terms aggregation

    ```
    POST testindex/logs/_search
    {
       "size": 0,
       "aggs": {
          "snlist": {
             "terms": {
                "field": "snlist"
             }
          }
       }
    }

   {
   "took": 8,
   "timed_out": false,
   "_shards": {
      "total": 1,
      "successful": 1,
      "failed": 0
   },
   "hits": {
      "total": 1,
      "max_score": 0,
      "hits": []
   },
   "aggregations": {
      "snlist": {
         "doc_count_error_upper_bound": 0,
         "sum_other_doc_count": 0,
         "buckets": [
            {
               "key": "106614,106607",
               "doc_count": 1
            }
         ]
      }
   }
   } 
    ```

### PR DESCRIPTION
**Pull request by johtani**
Remove request params in _analyze API without index param
Change rest-api-test using JSON
Change docs using JSON

Closes #20246

### CODE REVIEW

**Comment by johtani** [comment]
I will add breaking changes in migration doc.

File: core/src/test/java/org/elasticsearch/rest/action/admin/indices/RestAnalyzeActionTests.java [hunk 1]
```diff
@@ -19,16 +19,25 @@
...
+import org.elasticsearch.rest.RestRequest;
 import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.rest.FakeRestRequest;
+
+import java.util.HashMap;
 
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.notNullValue;
 import static org.hamcrest.Matchers.startsWith;
+import static org.mockito.Mockito.doCallRealMethod;
```
Comment (dadoonet) [inline core/src/test/java/org/elasticsearch/rest/action/admin/indices/RestAnalyzeActionTests.java]: Lot of imports have been added here. Are they really needed?

**Review by dadoonet — APPROVED** [review]
I left a small comment about imports.

BTW, thinking out loud and not really related to your PR, I wonder if we should document the `GET` verb instead of using `POST` as we are doing for `_search` endpoint. We don't change the server state when calling `_analyze` so `GET` makes more sense to me from a REST semantic point of view.

File: core/src/test/java/org/elasticsearch/rest/action/admin/indices/RestAnalyzeActionTests.java (hunk 1 above)
Comment (johtani) [inline core/src/test/java/org/elasticsearch/rest/action/admin/indices/RestAnalyzeActionTests.java]: Good catch! That is no need. I will check and fix.

**Comment by johtani** [comment]
@dadoonet Thx for reviewing. I fixed the issues you mentioned and add breaking changes.

File: docs/reference/migration/migrate_6_0/rest.asciidoc [hunk 2]
```diff
@@ -7,3 +7,7 @@ In previous versions of Elasticsearch, JSON documents were allowed to contain un
 This feature was removed in the 5.x series, but a backwards-compability layer was added via the
 system property `elasticsearch.json.allow_unquoted_field_names`. This backwards-compability layer
 has been removed in Elasticsearch 6.0.0.
+
+==== Analyze API changes
+
+The deprecated request parameters and plain text in request body has been removed, use JSON in request body.
```
Comment (dadoonet) [inline docs/reference/migration/migrate_6_0/rest.asciidoc]: "...have been removed. Use..." instead.

**Review by dadoonet — APPROVED** [review]
Left a minor comment. It looks good to me.

**Comment by johtani** [comment]
@clintongormley Question about rest-api-spec. In this PR, I have removed request params. Should I remove the params in [rest-api-spec](https://github.com/elastic/elasticsearch/blob/9eb242a5fedfe10af8dc208e63f952a4917c5af9/rest-api-spec/src/main/resources/rest-api-spec/api/indices.analyze.json#L14) ?
I'm not sure the difference between JSON param and request params in the spec.

**Comment by javanna** [comment]
@johtani we don't document in our spec what should go within the request body, we only define whether a body is required or not. The `params` section in the spec is for query_string parameters, which are what you removed, hence that section should be adapted ;)

File: docs/reference/migration/migrate_6_0/rest.asciidoc (hunk 2 above)
Comment (javanna) [inline docs/reference/migration/migrate_6_0/rest.asciidoc]: can we also be more generic when mentioning the request body? We don't support only the json format.

**Comment by johtani** [comment]
@javanna Thanks for you comment. I removed params in spec json file and updated REST changes message.

**Comment by johtani** [comment]
Rebased master

**Review by javanna — APPROVED** [review]
LGTM

**Comment by johtani** [comment]
@javanna @dadoonet Thx for the review!

**Pull request merged**